pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -170,10 +170,36 @@ func CommenceUpgrade(c client.Client, cfg *osdUpgradeConfig, scaler scaler.Scale
 	if err != nil {
 		return false, err
 	}
+
+	// Move the cluster to the same channel first
+	if cv.Spec.Channel != desired.Channel {
+		logger.Info(fmt.Sprintf("Moving cluster from Channel %s to Channel %s", cv.Spec.Channel, desired.Channel))
+		cv.Spec.Channel = desired.Channel
+		err = c.Update(context.TODO(), cv)
+		if err != nil {
+			return false, err
+		}
```

Comment:
Does https://github.com/openshift/managed-upgrade-operator/pull/101/files#diff-874a795bdc15ea8017a71b1b80606326R197 ?

Would it make more sense to `return false, nil` after changing the channel?

Reply:
Would you believe I originally had it doing exactly that (just returning `false, err` actually), then I changed my mind. 
But it's probably cleaner with the original approach. Addressed in latest commit.

Reply:
I believe :1st_place_medal: